### PR TITLE
chore(infra): optimize github tmpl & update actions/checkout version to 4.1.7

### PR DESCRIPTION
**Pull request by yuluo-yx**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+blank_issues_enabled: false
+contact_links:
+  - name: Community Support
+    url: https://github.com/apache/fury/discussions
+    about: Please ask and answer questions here.
+  - name: Security Bug Bounty
```
Comment (chaokunyang) [inline .github/ISSUE_TEMPLATE/config.yml:23]: Security report should not be here. Reporter should follow https://github.com/apache/.github/blob/main/.github/SECURITY.md

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+title: "[<Fury component: Protocol|Java|Python|C++|JavaScript|etc...>] "
+assignees: ''
+labels: ["new feature"]
+
+body:
+  - type: textarea
+    attributes:
+      label: Feature Request
+      description: >
+        Tip: Have you checked the docs https://hertzbeat.apache.org/docs/, GitHub issues whether someone else has already reported your issue? Maybe the feature already exists?
```
Comment (chaokunyang) [inline .github/ISSUE_TEMPLATE/feature_request.yml:29]: Why hertzbeat here

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -48,7 +48,7 @@ jobs:
       matrix:
         java-version: ["8", "11", "17", "21"]
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # 4.1.7
```
Comment (chaokunyang) [inline .github/workflows/ci.yml:51]: Why use this specific commit

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (yuluo-yx) [reply .github/ISSUE_TEMPLATE/feature_request.yml:29]: I will update in next commit

File: .github/workflows/ci.yml (hunk 3 above)
Comment (yuluo-yx) [reply .github/workflows/ci.yml:51]: More granular control over version numbers. If not necessary, I will update to @v4 version

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (yuluo-yx) [reply .github/ISSUE_TEMPLATE/config.yml:23]: got it

File: .github/workflows/ci.yml (hunk 3 above)
Comment (yuluo-yx) [reply .github/workflows/ci.yml]: done!

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (yuluo-yx) [reply .github/ISSUE_TEMPLATE/config.yml:23]: done

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    url: https://github.com/apache/.github/blob/main/.github/SECURITY.md
+    about: Please report security vulnerabilities here.
```
Comment (chaokunyang) [inline .github/ISSUE_TEMPLATE/config.yml:25]:
```suggestion
    about: To report a new vulnerability you have discovered please follow the ASF vulnerability reporting process.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+blank_issues_enabled: false
+contact_links:
+  - name: Community Support
+    url: https://github.com/apache/fury/discussions
+    about: Please ask and answer questions here.
+  - name: Security Bug Bounty
+    url: https://github.com/apache/.github/blob/main/.github/SECURITY.md
```
Comment (chaokunyang) [inline .github/ISSUE_TEMPLATE/config.yml:24]:
```suggestion
    url: https://apache.org/security/#reporting-a-vulnerability
```

**Review by chaokunyang — APPROVED** [review]
LGTM

**Comment by chaokunyang** [comment]
image
seems this is duplciate, could we remove one of those?

**Pull request merged**